packages/core/src/rest/Youtube.ts

```diff
@@ -134,7 +134,7 @@ export async function trackSearchByString(query: string, omitStreamId?: string,
     };
   }
     catch (e){
-      logger.error(`Warning: topTrack.url is undefined, removing song`);
+      throw new Error(`Warning: topTrack.url is undefined, removing song`);
       return undefined;
```

Comment:
You don't need `return` when u `throw` an error and log the original error is ok. Like this:
```js
    logger.error('youtube track search error');
    logger.error(e);
    throw new Error(`Warning: topTrack.url is undefined, removing song`);
```

Reply:
your recommendations have been added